posts/2022-06-30-Rust-1.62.0.md

```diff
@@ -0,0 +1,122 @@
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.62.0][notes] on GitHub.
+
+If you'd like to help us out by testing future releases, you might consider updating locally to use
+the beta channel (`rustup default beta`) or the nightly channel (`rustup default nightly`).
+Please [report] any bugs you might come across!
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1620-2022-06-30
+[report]: https://github.com/rust-lang/rust/issues/new/choose
+
+## What's in 1.62.0 stable
+
+### `cargo add`
+
+You can now add new dependencies directly from the command line using `cargo add`. This command supports specifying features and versions. It can also be used to modify existing dependencies.
+
+For example:
+
+```console
+cargo add log
+cargo add serde --features derive
+cargo add nom@5
+```
+
+See the [cargo documentation](https://doc.rust-lang.org/nightly/cargo/commands/cargo-add.html) for more.
+
+### `#[default]` enum variants
+
+You can now use `#[derive(Default)]` on enums if you specify a default variant. For example, until now you would have to manually write a `Default` impl for this enum:
+
+
+```rust
+#[derive(Default)]
+enum Maybe<T> {
+    #[default]
+    Nothing,
+
+    Something(T),
+}
+```
+
+As of now only "unit" variants (variants that have no fields) are allowed to be marked `#[default]`. More information is available in the [RFC](https://rust-lang.github.io/rfcs/3107-derive-default-enum.html) for this feature.
+
+### Thinner, faster mutexes on Linux
+
+Previously, `Mutex`, `Condvar`, and `RwLock` were backed by the pthreads library on Linux. The pthreads locks support more features than the Rust APIs themselves do, including runtime configuration, and are designed to be used in languages with fewer static guarantees than Rust provides.
+
+The mutex implementation, for example, is 40 bytes and cannot be moved. This forced the standard library to allocate a `Box` behind the scenes for each new mutex for platforms that use pthreads.
+
+Rust's standard library now ships with a raw futex-based implementation of these locks on Linux, which is very lightweight and doesn't require extra allocation. In 1.62.0 `Mutex` only needs 5 bytes for its internal state on Linux, though this may change in future versions.
+
+This is part of a long effort to improve the efficiency of Rust's lock types, which includes previous improvements on Windows such as unboxing its primitives. You can read more about that effort in the [tracking issue](https://github.com/rust-lang/rust/issues/93740).
+
+### Bare metal `x86_64` target
+
+It's now easier to build OS-less binaries for `x86_64`, for example when writing a kernel. `x86_64-unknown-none` has been promoted to a Tier 2 target and can be installed with rustup.
```

Comment:
I have we can link https://doc.rust-lang.org/beta/rustc/platform-support/x86_64-unknown-none.html and perhaps the target tier policy from tier 2.